src/App.js

```diff
@@ -25,16 +25,22 @@ LogBox.ignoreLogs([
 const App = () => (
     <ComposeProviders
         components={[
-            OnyxProvider,
             SafeAreaProvider,
             SafeArea,
             LocaleContextProvider,
-            HTMLEngineProvider,
         ]}
     >
         <CustomStatusBar />
+
         <ErrorBoundary errorMessage="NewExpensify crash caught by error boundary">
-            <Expensify />
+            <ComposeProviders
```

Comment:
As an error could happen on context provider initialization, `ErrorBoundary` should wrap them to hide the splash screen is something goes wrong.

Reply:
If there's something that might cause an Error in these components can you point it out? 

The parameters used in these components are static, so if an error is to happen it would happen all the time. Meaning if someone introduces and error they'll see it right away

If this change is accepted we should at least add a reminder doc in `ErrorBoundary` that certain context based functionality is not available inside

Though I would prefer to keep the error boundary access the full context (e.g. `withNetwork`, `withPersonalDetails` and others), and only relinquish it if there's a proven possibility that certain context can cause an error

Reply:
`react-native-onyx` might have a bug, no software is perfect. To display an error fallback, it's not worth it to try (imho)

Reply:
Yeah, the code in `ErrorBoundary` might cause an error as well, we're obviously not adding another `ErrorBoundary` to guard from that

In the future more context components will be created it would be easier if those are kept on the same place and on the same level 
- people won't wonder where to put them 
- people would not mistakenly put them to the incorrect place

In the feature more components will be created and some of these component would rely on context, 
Adding such a component (Like a network status indicator) in the error boundary would not work for seemingly no reason

If there's a bug in the initialization of some Context it fail with a bang any way and never make it to production. Wrapping the error boundary around the context is not necessary at least the currently defined contexts

---

To sum it up Moving the `ErrorBoundary` up would work, but introduces restrictions, while it's uncertain that it will help
Not changing the Error boundary position would work as well, and it's not necessary to move it to fix the splash screen bug

Reply:
I agree with @kidroca here, this change is accounting for an error that we haven't even encountered yet and it has larger implications than just catching potential initialization errors.